bundles/org.platformio.eclipse.ide.home/src/org/platformio/eclipse/ide/home/json/Dump.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ * Copyright (c) 2021 PlatformIO and ArSysOp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Contributors:
+ *     Nikifor Fedorov (ArSysOp) - initial API and implementation
+ *******************************************************************************/
+package org.platformio.eclipse.ide.home.json;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.platformio.eclipse.ide.home.python.Python;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+
+public final class Dump {
+
+	private final Path envBinDir;
+	private final Path envDir;
+
+	public Dump() throws IOException {
+		BundleContext context = FrameworkUtil.getBundle(getClass()).getBundleContext();

```

Comment:
"no code in constructor" principle is viloated here
there should be a caching function to resolve this, see `ru.arsysop.lang` library